flink-core/src/main/java/org/apache/flink/core/fs/local/LocalDataOutputStream.java

```diff
@@ -47,31 +52,49 @@ public LocalDataOutputStream(final File file) throws IOException {
 
     @Override
     public void write(final int b) throws IOException {
+        mayThrowClosedChannelException();
+        fos.write(b);
+    }
+
+    @Override
+    public void write(@Nonnull final byte[] b) throws IOException {
+        mayThrowClosedChannelException();
         fos.write(b);
     }
 
     @Override
     public void write(final byte[] b, final int off, final int len) throws IOException {
+        mayThrowClosedChannelException();
         fos.write(b, off, len);
     }
 
     @Override
     public void close() throws IOException {
         fos.close();
+        isClosed = true;
```

Comment:
The intention was that we don't flag it if there's an error in `fos.close()` because we don't know, whether the closing actual happen. But thinking about it once more, you might be right: The caller had the intention of closing the `OutpuStream`, i.e. no call accessing the stream should be possible anymore in the caller code even if an error occurred. 👍